django_prbac/utils.py

```diff
@@ -6,40 +6,37 @@
 from django_prbac.models import Role, UserRole
 
 
-def ensure_request_has_privilege(request, slug, **assignment):
+def has_privilege(request, slug, **assignment):
     """
-    Ensures that an HttpRequest object has the privilege specified by slug.
-    If it does not, it throws a PermissionDenied Exception.
+    Returns true if the request has the privilege specified by slug,
+    otherwise false
     """
     if not hasattr(request, 'role'):
-        raise PermissionDenied()
+        return False
 
-    roles = Role.objects.filter(slug=slug)
-    if not roles:
-        raise PermissionDenied()
+    privilege = Role.get_privilege(slug, assignment)
+    if privilege is None:
+        return False
 
-    privilege = roles[0].instantiate(assignment)
     if request.role.has_privilege(privilege):
-        return
+        return True
 
     if not hasattr(request, 'user') or not hasattr(request.user, 'prbac_role'):
-        raise PermissionDenied()
-
+        return False
     try:
         request.user.prbac_role
     except UserRole.DoesNotExist:
-        raise PermissionDenied()
+        return False
 
-    if not request.user.prbac_role.has_privilege(privilege):
-        raise PermissionDenied()
+    return request.user.prbac_role.has_privilege(privilege)
 
 
-def has_privilege(request, slug, **assignment):
+def ensure_request_has_privilege(request, slug, **assignment):
     """
-    Returns true if the request has hte privilege, otherwise false
+    DEPRECATED
+
+    You most likely want `has_permission` or one of the
+    `requires_privilege` decorators.
```

Comment:
You should use [warnings](https://docs.python.org/2/library/warnings.html) to indicate deprecation